src/infill/SpaghettiInfill.cpp

```diff
@@ -0,0 +1,139 @@
+/** Copyright (C) 2017 Ultimaker - Released under terms of the AGPLv3 License */
+#include "SpaghettiInfill.h"
+
+namespace cura {
+
+
+void SpaghettiInfill::generateSpaghettiInfill(SliceMeshStorage& mesh)
+{
+    coord_t layer_height = mesh.getSettingInMicrons("layer_height");
+    coord_t line_width = mesh.getSettingInMicrons("infill_line_width");
+    double layer_height_mm = INT2MM(layer_height);
+    int spaghetti_max_layer_count = std::max(1, static_cast<int>(mesh.getSettingInMicrons("spaghetti_max_height") / layer_height));
+    // TODO: account for the initial layer height
```

Comment:
took quite some refactoring, but I did it.

pillars now record volume rather than area and bottom z coord rather than layer_count